storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -4416,6 +4454,25 @@ void xtrabackup_backup_func(void) {
     exit(EXIT_FAILURE);
   }
 
+  if (xtrabackup_fifo_streams_set && xtrabackup_fifo_dir != NULL) {
+    xtrabackup_target_dir = xtrabackup_fifo_dir;
```

Comment:
what?? we change target_dir to fifo_dir?

Reply:
shouldn't it be opposite if fifo_streams is set and a directory is not passed, we should use target_dir as fifo_dir?

Reply:
That is what we do. `xtrabackup_fifo_streams_set = --fifo-streams > 1` `xtrabackup_fifo_dir = --fifo-dir missing`

Basically, if you are using `--fifo-streams=2` you can save the FIFO files into `--fifo-dir`, however, for convenience you don't need to remember yet another parameter, you can use `--fifo-streams=2 --target-dir=path` and fifo files will be written on path.